src/Navigation/ValidateConditions.ts

```diff
@@ -0,0 +1,44 @@
+/**
+ * @Author: Erik Slovak <erik.slovak@technologystudio.sk>
+ * @Date: 2022-11-08T08:11:68+01:00
+ * @Copyright: Technology Studio
+**/
+
+import { conditionalNavigationManager } from '../Api/ConditionalNavigationManager'
+import {
+  getActiveLeafRoute,
+  getActiveRoutePath,
+  getScreenNavigationConditions,
+} from '../Api/NavigationUtils'
+import type {
+  OnActionAttributes,
+  ResolveConditionsResult,
+  ValidateConditionsNavigationAction,
+} from '../Model/Types'
+
+export const onValidateConditionsAction = ({
+  action,
+  getContext,
+  getState,
+  originalOnAction,
+  restArgs,
+  screenConditionConfigMap,
+}: OnActionAttributes<ValidateConditionsNavigationAction>): boolean => {
+  const state = getState()
+  const currentActiveScreenPath = getActiveRoutePath(state) ?? []
+  if (state) {
+    let resolveConditionsResult: ResolveConditionsResult | undefined
+    for (const routeName of currentActiveScreenPath) {
+      const screenConditions = getScreenNavigationConditions(screenConditionConfigMap[routeName])
+      if (screenConditions && screenConditions.length > 0) {
+        resolveConditionsResult = conditionalNavigationManager.resolveConditions(screenConditions, action, state, getContext)
+      }
+    }
```

Comment:
Here we have the same issue with ignored result if there are 3 routes, and 1 and 3 has conditions and there is an result, so first result is ignored.